apps/emqx/src/emqx_trace/emqx_trace_formatter.erl

```diff
@@ -89,8 +89,17 @@ weight({packet, _}) -> {0, packet};
 weight({payload, _}) -> {2, payload};
 weight({K, _}) -> {1, K}.
 
-format_packet(undefined, _) -> "";
-format_packet(Packet, Encode) -> emqx_packet:format(Packet, Encode).
+format_packet(undefined, _) ->
+    "";
+format_packet(Packet, Encode) ->
+    try
+        emqx_packet:format(Packet, Encode)
+    catch
+        _:_ ->
+            %% We don't want to crash if there is a field named packet with
```

Comment:
could you provide an example here?

Reply:
There is a trace that includes the output of a rule. As the the rule can output almost any map it can contain a field named packet. Not 100% sure if the key can be an atom in that case. Perhaps the optimal solution is to not look for the packet field in all levels of the map as we may only want to format it when it is in the top level. What do you think? The payload field is perhaps different since the user can change how it is formatted and will perhaps expect it to be formatted this way wherever it may be.

Reply:
this is ok.
I missed that this is done recursively.
maybe good to explain as comment.